src/scripts/saveToOneNote/oneNoteSaveableFactory.ts

```diff
@@ -172,6 +173,12 @@ export class OneNoteSaveableFactory {
 
 	private static getPageIndicesToSendInPdfMode(clipperState: ClipperState) {
 		let pdf = clipperState.pdfResult.data.get().pdf;
-		return clipperState.pdfPreviewInfo.allPages ? _.range(pdf.numPages()) : StringUtils.parsePageRange(clipperState.pdfPreviewInfo.selectedPageRange, pdf.numPages()).map(value => value - 1);
+		const parsePageRangeOperation = StringUtils.parsePageRange(clipperState.pdfPreviewInfo.selectedPageRange, pdf.numPages());
+		if (parsePageRangeOperation.status !== Status.Succeeded) {
+			throw new Error("User is clipping an invalid page range");
+		}
+		const pageRange = parsePageRangeOperation.result as number[];
+
+		return clipperState.pdfPreviewInfo.allPages ? _.range(pdf.numPages()) : pageRange.map(value => value - 1);
```

Comment:
Check allPages and return the _.range(pdf.numPages()) at the beginning so you don't need to compute the pageRange result every time.